esx_service/utils/auth_api.py

```diff
@@ -0,0 +1,296 @@
+import auth
+import auth_data_const
+import convert
+import auth_data
+import vmdk_utils
+
+def get_auth_mgr():
+    try:
+        auth_mgr = auth.get_auth_mgr()
+    except auth_data.DbConnectionError, e:
+        error_info = "Failed to connect auth DB({0})".format(e)
+        return error_info, None
+    return None, auth_mgr
+
+def get_tenant_from_db(name):
+    error_info, auth_mgr = get_auth_mgr()
+    if error_info:
+        return error_info, None
+
+    error_info, tenant = auth_mgr.get_tenant(name)
+    return error_info, tenant
+
+def create_tenant_in_db(name, description, default_datastore, default_privileges, vms, privileges):
+    error_info, auth_mgr = get_auth_mgr()
+    if error_info:
+        return error_info, None
+
+    error_info, tenant = auth_mgr.create_tenant(name = name, 
+                                                description = description, 
+                                                default_datastore = default_datastore, 
+                                                default_privileges = default_privileges, 
+                                                vms = vms, 
+                                                privileges = privileges)
+    return error_info, tenant
+
+def get_tenant_list_from_db():
+    error_info, auth_mgr = get_auth_mgr()
+    if error_info:
+        return error_info, None
+    
+    error_info, tenant_list = auth_mgr.list_tenants()
+    return error_info, tenant_list
+
+def generate_tuple_from_vm_list(vm_list):
+    """ Generate a list of (vm_uuid, vm_name) pair """
+    if not vm_list:
+        return None, []
+    vms = []
+    for vm_name in vm_list:
+        vm_uuid = vmdk_utils.get_vm_uuid_by_name(vm_name)
+        if not vm_uuid:
+            error_info =  "Cannot find vm_uuid for vm {0}, tenant_create failed".format(vm_name)
+            return error_info, None
```

Comment:
I think I had commented on this last PR. We should not terminate loop on first bad VM name. Rather iterate through entire list and return VMs that are successfully assigned and VMs that couldn't be found.